In [2]:
import pandas as pd
import numpy as np

In [11]:
production_materials={'product_quantity':[7,4,5,9,8,5,7,7,6,8],
                     'product_quality':[2,4,3,4,3,5,5,2,3,4],
                     'product_cost':[2300,3200,1899,3019,2056,3010,2890,2650,2999,3010]}
df=pd.DataFrame(production_materials)
df

product_quantity  product_quality  product_cost
0                 7                2          2300
1                 4                4          3200
2                 5                3          1899
3                 9                4          3019
4                 8                3          2056
5                 5                5          3010
6                 7                5          2890
7                 7                2          2650
8                 6                3          2999
9                 8                4          3010

# Finding the average of all the features

In [12]:
df['product_quantity'].mean()

6.6

The average quantity of the products is around 6.6

In [14]:
df['product_quality'].mean()

3.5

The average quality rating of the products is 3.5

In [15]:
df['product_cost'].mean()

2703.3

The average cost of the product is 2703.3

# Finding the most common value and the next common value

In [25]:
count =df['product_quantity'].value_counts()
count

product_quantity
7    3
5    2
8    2
4    1
9    1
6    1
Name: count, dtype: int64

In [28]:
diff=count.index[0]-count.index[1]
diff

2

7 is  most common quantity of the product with 3 times and the the next most common product quantity is 5,8 with 2 times.
The differance between most common quantity of product(7) and the next common quantity (5,8) is 2